program/src/instruction.rs

```diff
@@ -1,184 +1,420 @@
+    #[account(
+        6,
+        writable,
+        name = "vault",
+        desc = "Stake token Vault"
+    )]
+    #[account(
+        7,
+        name = "spl_token_program",
+        desc = "SPL Token 2022 program"
+    )]
     StakeTokens(u64),
 
     /// Deactivate staked tokens for the validator.
-    ///
+    /// 
     /// Only one deactivation may be in-flight at once, so if this is called
     /// with an active deactivation, it will succeed, but reset the amount and
     /// timestamp.
-    ///
-    /// 0. `[w]` Validator stake account
-    /// 1. `[s]` Authority on validator stake account
-    ///
-    /// Instruction data: amount of tokens to deactivate, as a little-endian u64
+    /// 
+    /// Instruction data: amount of tokens to deactivate, as a little-endian `u64``.
+    #[account(
+        0,
+        writable,
+        name = "stake",
+        desc = "Validator stake account"
+    )]
+    #[account(
+        1,
+        signer,
+        name = "stake_authority",
+        desc = "Authority on validator stake account"
+    )]
     DeactivateStake(u64),
 
     /// Move tokens from deactivating to inactive.
-    ///
+    /// 
     /// Reduces the total voting power for the stake account and the total staked
     /// amount on the system.
-    ///
+    /// 
     /// NOTE: This instruction is permissionless, so anybody can finish
     /// deactivating someone's tokens, preparing them to be withdrawn.
-    ///
-    /// 0. `[w]` Stake config account
-    /// 1. `[w]` Validator stake account
+    #[account(
+        0,
+        writable,
+        name = "config",
+        desc = "Stake config account"
+    )]
+    #[account(
+        1,
+        writable,
+        name = "stake",
+        desc = "Validator stake account"
+    )]
     InactivateStake,
 
-    /// Withdraw inactive staked tokens from the vault
-    ///
+    /// Withdraw inactive staked tokens from the vault.
+    /// 
     /// After a deactivation has gone through the cooldown period and been
     /// "inactivated", the authority may move the tokens out of the vault.
-    ///
-    /// 0. `[w]` Config account
-    /// 1. `[w]` Stake account
-    /// 2. `[w]` Vault token account
-    /// 3. `[w]` Destination token account
-    /// 4. `[s]` Stake authority
-    /// 5. `[]` Vault authority, PDA with seeds `['token-owner', stake_config]`
-    /// 6. `[]` SPL Token program
-    /// 7.. Extra required accounts for transfer hook
-    ///
-    /// Instruction data: amount of tokens to move
+    /// 
+    /// Instruction data: amount of tokens to move.
+    #[account(
+        0,
+        writable,
+        name = "config",
+        desc = "Stake config account"
+    )]
+    #[account(
+        1,
+        writable,
+        name = "stake",
+        desc = "Stake account"
+    )]
+    #[account(
+        2,
+        writable,
+        name = "vault",
+        desc = "Vault token account"
+    )]
+    #[account(
+        3,
+        writable,
+        name = "destination_token",
```

Comment:
nit: can this be called `destination` or `destination_token_account`?

Reply:
I used `destination_token_account` since harvest instructions use `destination` to specify the account that receives the SOL rewards.